src/robot_arm.cpp

```diff
@@ -87,22 +199,53 @@ char *RobotArm::strcopy(char *dest, const char *src) {
     while (*src) {
         *dest++ = *src++;
     }
+
     *dest++ = '\0';
-    *dest = 0;
+
     return saved;
 }
 
 char *RobotArm::intToChar(int number, char *dest) {
-    if (number / 10 == 0) {
+    if ((number / 10) == 0) {
         *dest++ = number + '0';
         *dest = '\0';
         return dest;
     }
 
     dest = intToChar(number / 10, dest);
-    *dest++ = number % 10 + '0';
+    *dest++ = (number % 10) + '0';
     *dest = '\0';
     return dest;
 }
 
-}
\ No newline at end of file
+int RobotArm::charToInt(const char *str, const unsigned int posStart, const unsigned int posEnd) const {
+    int result = 0;
+
+    for (unsigned int i = posStart; i < posEnd; i++) {
+        char digit = static_cast<char>(str[i] - '0');
+
+        result *= 10;
+        result += digit;
+    }
+
+    return result;
+}
+
+int RobotArm::getCharPositionStr(const char *str, const char search, const int searchStart) const {
+    unsigned int strIndex = searchStart;
+
+    const char *strSearchStart = str + searchStart;
+
+    while (*strSearchStart != '\0') {
+        if (*strSearchStart == search) {
+            return strIndex;
+        }
+
+        strSearchStart++;
```

Comment:
I personally prefer ++var instead of var++

Reply:
I like it the other way ;)